urbs.py

```diff
@@ -927,17 +1085,59 @@ def split_columns(columns, sep='.'):
 
 def commodity_subset(com_tuples, type_name):
     """ Unique list of commodity names for given type. 
-    
+
     Args:
         com_tuples: a list of (site, commodity, commodity type) tuples
         type_name: a commodity type ('Stock', 'SupIm', 'Env' or 'Demand')
-        
+
     Returns:
         The set (unique elements) of commodity names of the desired type
     """
     return set(com for sit, com, com_type in com_tuples 
                if com_type == type_name)
 
+def commodity_subset_list(com_tuples, type_list):
+    """ Unique list of commodity names for given list (type).
+
+    Args:
+        com_tuples: a list of (site, commodity, commodity type) tuples
+        type_list: list of a commodity type ('Buy')=>('Elec buy', 'Heat buy')
+
+    Returns:
+        The set (unique list of tuples) of tuples of the desired type
+    """
+    return set((sit, com, com_type) for sit, com, com_type in com_tuples
+               if com in type_list)
+
+def extract_number_str(str_in):
+    """ Extract the first number from a given string and
+        and convert to the float type
+
+        Excel sheet: Commodity - price - for Buy and Sell:
+
+            - fix price: (0.12) [input type: float]
+                the same price for each timestep
+            - timeseries: (1.25BUY) [input type: string]
+                1.) extract the number 1.25
+                2.) select the com price for the given timestep
+                3.) calculate the price with the factor (1.25)
+                    => 1.25 x Buy-Sell-Price(h)
+
+    Args:
+        str_in: a string ('1,20BUY')
+
+    Returns:
+        A float number (1.20)
+    """
+    import re
+    # replace all commas with points
+    str_in = str_in.replace(',','.')
```

Comment:
This might break very easily if the locale of the spreadsheet application is different.